### PR TITLE
UN-3257 add new test case llm gemini

### PR DESCRIPTION
**Pull request by vince-leaf**
This PR is a new test case to test and confirm our neuro-san support for the Gemini model.

### CODE REVIEW

File: tests/fixtures/music_nerd_pro_llm_gemini/combination_responses_with_history_direct.hocon [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+
+# This file defines everything necessary for a data-driven test.
+{
+    # Describes what agent to test against.
+    "agent": "music_nerd_pro_llm_gemini",
```
Comment (vince-leaf) [inline tests/fixtures/music_nerd_pro_llm_gemini/combination_responses_with_history_direct.hocon:15]: Using the music_nerd_pro fixture for testing this LLM Gemini

File: tests/neuro_san/zzz_hocons/test_smoke_test_hocons.py [hunk 2]
```diff
@@ -91,22 +92,7 @@ def test_hocon_with_server(self, test_name: str, test_hocon: str):
...
-        :param test_hocon: The hocon file of a single data-driven test case.
-        """
-        # Call the guts of the dynamic test driver.
-        # This will expand the test_hocon file name from the expanded list to
-        # include the file basis implied by the __file__ and path_to_basis above.
-        self.DYNAMIC.one_test_hocon(self, test_name, test_hocon)
-    
-    @pytest.mark.gemini
-    def test_hocon_with_gemini(self, test_name: str, test_hocon: str):
+    def test_hocon_with_non_default_llm(self, test_name: str, test_hocon: str):
```
Comment (vince-leaf) [inline tests/neuro_san/zzz_hocons/test_smoke_test_hocons.py:95]: Renamed the default test name to be a bit more general, which is used for non_default_llm.

File: tests/neuro_san/zzz_hocons/test_smoke_test_hocons.py [hunk 3]
```diff
@@ -83,6 +83,7 @@ def test_hocon_with_server(self, test_name: str, test_hocon: str):
         # Ideally more basic functionality will come first.
         # Barring that, try to stick to alphabetical order.
         "music_nerd_pro_llm_anthropic/combination_responses_with_history_direct.hocon",
+        "music_nerd_pro_llm_gemini/combination_responses_with_history_direct.hocon",
```
Comment (vince-leaf) [inline tests/neuro_san/zzz_hocons/test_smoke_test_hocons.py:86]: The new test case triggered by the smoke test was added as a direct connection.

File: tests/conftest.py [hunk 4]
```diff
@@ -16,6 +17,9 @@ def configure_llm_provider_keys(request, monkeypatch):
         if is_anthropic:
             if not os.getenv("ANTHROPIC_API_KEY"):
                 pytest.skip("Missing ANTHROPIC_API_KEY for test marked 'anthropic'")
+        elif is_gemini:
+            if not os.getenv("GOOGLE_API_KEY"):
+                pytest.skip("Missing AGOOGLE_API_KEY for test marked 'anthropic'")
```
Comment (vince-leaf) [inline tests/conftest.py]: Added this condition to satisfy the @donn-leaf GitHub Action smoke test requirement.

File: pytest.ini [hunk 5]
```diff
@@ -6,6 +6,7 @@ markers =
     smoke: Mark a test as an smoke test. These often exercise the full stack
     non_default_llm_provider: Tests that exercise llm providers other than our default Open AI
     anthropic: Tests that specifically use anthropic as the llm provider
+    gemini: Tests that specifically use gemini as the llm provider
```
Comment (vince-leaf) [inline pytest.ini:9]: Ignored Pytest warning message.

File: neuro_san/registries/music_nerd_pro_llm_gemini.hocon [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+
+# The schema specifications for this file are documented here:
+# https://github.com/cognizant-ai-lab/neuro-san/blob/main/docs/agent_hocon_reference.md
+
+{
+    "llm_config": {
+        "model_name": "gemini-2.0-flash",
```
Comment (vince-leaf) [inline neuro_san/registries/music_nerd_pro_llm_gemini.hocon:18]: Added new agent

File: tests/neuro_san/zzz_hocons/test_smoke_test_hocons.py [hunk 7]
```diff
@@ -90,6 +90,7 @@ def test_hocon_with_server(self, test_name: str, test_hocon: str):
     @pytest.mark.smoke
     @pytest.mark.non_default_llm_provider
     @pytest.mark.anthropic
+    @pytest.mark.gemini
```
Comment (vince-leaf) [inline tests/neuro_san/zzz_hocons/test_smoke_test_hocons.py:94]: This is a requirement by the GitHub Action smoke test env

Comment (donn-leaf) [reply tests/neuro_san/zzz_hocons/test_smoke_test_hocons.py:94]: This part doesn't feel right, seems like mark.anthropic should only be in the anthropic test, and mark.gemini should be only in the gemini test. Unless I'm not understanding how these marks work.

File: tests/conftest.py [hunk 8]
```diff
@@ -17,7 +17,7 @@ def configure_llm_provider_keys(request, monkeypatch):
         if is_anthropic:
             if not os.getenv("ANTHROPIC_API_KEY"):
                 pytest.skip("Missing ANTHROPIC_API_KEY for test marked 'anthropic'")
-        if is_gemini:
+        elif is_gemini:
```
Comment (vince-leaf) [inline tests/conftest.py:20]: Fixed typo logic on the elif statement.

**Comment by vince-leaf** [comment]
I will need help at this point, as the test case grpc is failing due to the response from the agent network being "None". 
======
`python -m neuro_san.client.agent_cli --agent music_nerd_pro --connection grpc                  SIGINT(2) ↵  4347  15:04:47


I can help with music-related inquiries.


To see the thinking involved with the agent:

    See any one of the files in /private/tmp/agent_thinking for agent network chat details.

Please enter your response ('quit' to terminate):
Who did yellow submarine?
Sending user_input Who did yellow submarine?

Response from agent network:
None`
=====

But I can see the response from the agent server log
`========== Calling Accountant ==========
args: {'running_cost': '0.00', 'origin': [{'tool': 'MusicNerdPro', 'instantiation_index': 1}, {'tool': 'Accountant', 'instantiation_index': 1}], 'origin_str': 'MusicNerdPro.Accountant'}
-----------------------
Accountant response:  {'running_cost': 3.0}
========== Done with Accountant ==========
{"message": "Request reporting: {
    "token_accounting": {
        "caveats": [
            "Each LLM Branch Node also includes accounting for each of its callees."
        ],
        "completion_tokens": 100,
        "prompt_tokens": 1320,
        "successful_requests": 2,
        "time_taken_in_seconds": 5.781114101409912,
        "total_cost": 0.0043,
        "total_tokens": 1420
    }`

File: tests/neuro_san/zzz_hocons/test_smoke_test_hocons.py (hunk 2 above)
Comment (d1donlydfink) [reply tests/neuro_san/zzz_hocons/test_smoke_test_hocons.py:95]: This is OK for now. Just means piling all these tests together will mean that you will not be able to run them independently.  That is, if you have to run a gemini test, you will also have to run an anthropic test.
Maybe not so bad. We can still run specific test cases generated by the list via pytest.

**Review by d1donlydfink — APPROVED** [review]
